### PR TITLE
treydock's branch

### PR DESCRIPTION
**Pull request by ghoneycutt**

### CODE REVIEW

File: spec/classes/accesslogin_spec.rb [hunk 1]
```diff
@@ -22,26 +22,19 @@
...
-
-# allow only the groups listed
-+ : root : ALL
-
-# default deny
-- : ALL : ALL
-})
-      }
+      it "should contain default allowed_users" do
+        verify_contents(subject, 'access_conf', [
```
Comment (ghoneycutt) [inline spec/classes/accesslogin_spec.rb]: ?

File: manifests/init.pp [hunk 2]
```diff
@@ -47,7 +51,7 @@
 
   case $::osfamily {
     'RedHat': {
-      case $::lsbmajdistrelease {
+      case $::operatingsystemmajrelease {
```
Comment (ghoneycutt) [inline manifests/init.pp]: This can work in this use case since it is under RedHat. If used elsewhere it will break as Suse does not support the new fact.

**Comment by ghoneycutt** [comment]
@treydock your work is phenomenal! It is just too much all in one commit to just merge in. If you could split new functionality or bug fixes into their own pull requests, it is much easier to work together to get things merged. As it is now, I would need to cherry pick all the awesome around the few things I cannot merge or just do not understand enough to merge.

**Comment by treydock** [comment]
@ghoneycutt No problem, I'll try and make individual branches for each change and open up a PR per branch.

**Comment by treydock** [comment]
@ghoneycutt Some of my changes have been rushed and not very good for a module used by the community.  Do you see any changes I've made that would benefit this module that haven't already been merged?  I know my method of unit testing is different, but for actual functionality of the module I'm having trouble identifying what else could be of use.

**Comment by ghoneycutt** [comment]
We have done a major version upgrade to support Puppet 5 and 6. This involves the transition to hiera data in modules, acceptance testing, type validation in parameters and more.

Your contribution is greatly appreciated! Please check it out the new version 3 and re-open if needed.

**Pull request closed without merging**